hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationExecutor.java

```diff
@@ -136,26 +138,30 @@
      * @throws java.lang.NullPointerException if op is null.
      */
     @Deprecated
-    boolean isAllowedToRunInCurrentThread(Operation op);
+    boolean runAllowed(Operation op);
+
+    /**
+     * Checks if this operation is allowed to be invoked from the current thread. Invoking means that the operation can
+     * be executed on another thread, but that one is going to block for completion using the future.get/join etc.
+     * Blocking for completion can cause problems, e.g. when you hog a partition thread or deadlocks.
```

Comment:
Normally an invocation is done asynchronous; so you execute and you get a invocationfuture. But in most cases you immediately do a future.get so the calling thread is going to block for completion.

This methods checks if blocking for completion is allowed on the current thread for the given operation.

Does this make sense? Lets first make sure you understand it, and then we update the documentation accordingly.